src/winrm_client.py

```
@@ -0,0 +1,102 @@
+
+        # Upload the file
+        upload_script = f"""
+        try {{
+            $content = [System.Convert]::FromBase64String("{encoded_content}")
+            [System.IO.File]::WriteAllBytes("{remote_path}", $content)
+            if (Test-Path "{remote_path}") {{
+                Write-Output "Success"
+            }} else {{
+                Write-Output "Failed to write file to {remote_path}"
+            }}
+        }} catch {{
+            Write-Output "Failed: $($_.Exception.Message)"
+        }}
+        """
+
+        upload_result = self.session.run_ps(upload_script)
+        upload_output = upload_result.std_out.decode().strip()
+        upload_error = upload_result.std_err.decode().strip()
+
+        print(f"Upload script result: {upload_output}")
+        if upload_error:
+            print(f"Error uploading file: {upload_error}")
+
+        # Post-upload check
+        file_exists_check = f"Test-Path -Path '{remote_path}'"
+        file_exists_result = self.execute_command(file_exists_check).strip().lower()
+        print(f"File exists check result: {file_exists_result}")
+        if file_exists_result != "true":
+            raise FileNotFoundError(f"File {remote_path} does not exist after upload.")
+
+    def execute_command(self, command):
+        print(f"Executing command: {command}")
+        result = self.session.run_ps(command)
+        output = result.std_out.decode().strip()
+        error = result.std_err.decode().strip()
+        print(f"Command output: {output}")
+        if error:
+            print(f"Command error: {error}")
+        return output
+
+    def check_file(self, remote_path, local_checksum):
+        checksum_script = f"""
+        $file = Get-Item -Path "{remote_path}"
+        if ($file -ne $null) {{
+            $checksum = Get-FileHash -Path "{remote_path}" -Algorithm SHA256
+            $checksum.Hash
+        }} else {{
+            Write-Output "File does not exist"
```

Why not first check if the file exists, raise exception if file is not found and then get the checksum of the file?
Avoid using extra steps: writing-outputting "File does not exist" and parsing the output after